app/helpers/wash_out_helper.rb

```diff
@@ -39,7 +39,8 @@ def wsdl_data(xml, params)
               blk = proc { wsdl_data(xml, p.map) }
             end
             attrs.reject! { |_, v| v.nil? }
-            xml.tag! tag_name, param_options.merge(attrs), &blk
+            options_and_attrs = param_options.is_a?(Hash) ? param_options.merge(attrs) : attrs
```

Comment:
I think `param_options` is always a hash, because it contains whatever `wsdl_data_options`is returning, which should be either `{ :"xsi:type" => param.namespaced_type }`, `{ :"xsi:nil" => true }` or `{ }`

Reply:
Unfortunately that is not the case. We get errors that merge is not a method on nil class.

Reply:
You're right, 🤔  what about adding an `else` clause in the `wsdl_data_options` method? for when the `controller.soap_config.wsdl_style` is neither "rpc" or "document"? Like:
```
def wsdl_data_options(param)
  ...
  when 'document'
    { }
  else
    { }
  end
end
```

IMHO that will fix bouth lines 32 and 42 and keep consistency on the `wsdl_data_options` method.

Reply:
Done! Sorry about that, I didn't read the code for the `wsdl_data_options` well enough. Your suggestion has been verified to work for us in both scenarios.